packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -0,0 +1,91 @@
+import winston from "winston";
+
+import { createEtherscanLinkMarkdown, createFormatFunction, PublicNetworks } from "@uma/common";
+import { getAbi } from "@uma/contracts-node";
+import { InsuredBridgeL1Client, InsuredBridgeL2Client, GasEstimator } from "@uma/financial-templates-lib";
+
+export enum RelaySubmitType {
+  Slow,
+  SpeedUp,
+  Instant,
+  Ignore,
+}
+
+export class CrossDomainFinalizer {
+  constructor(
+    readonly logger: winston.Logger,
+    readonly gasEstimator: GasEstimator,
+    readonly l1Client: InsuredBridgeL1Client,
+    readonly l2Client: InsuredBridgeL2Client,
+    readonly account: string
+  ) {}
+  async checkForBridgeableL2TokensAndBridge() {
+    this.logger.debug({ at: "AcrossRelayer#CrossDomainFinalizer", message: "Checking bridgeable L2 tokens" });
+
+    const whitelistedTokenEvents = await this.l1Client.bridgeAdmin.getPastEvents("WhitelistToken", {
+      fromBlock: 0,
+      toBlock: await this.l1Client.l1Web3.eth.getBlockNumber(),
+    });
+
+    // Extract the l2Tokens that have been whitelisted. The new Set syntax acts to remove any duplicates from the array.
+    // This would be the case if a l2 token was re - whitelisted after upgrading the bridgePool contract on L1.
+    const whitelistedL2Tokens = [...new Set(whitelistedTokenEvents.map((event) => event.returnValues.l2Token))];
+
+    // Check if any of the whitelisted l2Tokens are bridgeable. Do this in one parallel call. Returns an array of bool
```

Comment:
Do you think we should add an `if` statement that checks if the bridge deposit box's balance is > some threshold? If its too small then perhaps we hold off bridging to save gas? I know L2 gas is negligible but still

Reply:
yup! I will change this to be greater than x% of the amount of liquid + utilised funds on L1. i.e the funds waiting on to be bridged on the canonical bridge are greater than 5% of the funds on L1 then, and only then, bridge